admin/js/novaramedia-audio-tool-admin.js

```diff
@@ -1,32 +1,137 @@
 (function( $ ) {
-	'use strict';
-
-	/**
-	 * All of the code for your admin-facing JavaScript source
-	 * should reside in this file.
-	 *
-	 * Note: It has been assumed you will write jQuery code here, so the
-	 * $ function reference has been prepared for usage within the scope
-	 * of this function.
-	 *
-	 * This enables you to define handlers, for when the DOM is ready:
-	 *
-	 * $(function() {
-	 *
-	 * });
-	 *
-	 * When the window is loaded:
-	 *
-	 * $( window ).load(function() {
-	 *
-	 * });
-	 *
-	 * ...and/or other possibilities.
-	 *
-	 * Ideally, it is not considered best practise to attach more than a
-	 * single DOM-ready or window-load handler for a particular page.
-	 * Although scripts in the WordPress core, Plugins and Themes may be
-	 * practising this, we should strive to set a better example in our own work.
-	 */
-
-})( jQuery );
+  'use strict';
+
+  // selectText helper
+  jQuery.fn.selectText = function() {
+    var doc = document
+      , element = this[0]
+      , range, selection
+    ;
+    if (doc.body.createTextRange) {
+      range = document.body.createTextRange();
+      range.moveToElementText(element);
+      range.select();
+    } else if (window.getSelection) {
+      selection = window.getSelection();
+      range = document.createRange();
+      range.selectNodeContents(element);
+      selection.removeAllRanges();
+      selection.addRange(range);
+    }
+  };
+
+  // AudioTool object
+  var AudioTool = {
+    vars: AudioToolVars,
+
+    init: function() {
+       var _this = this;
+
+       _this.bind();
+
+      $('.easy-select').click(function() {
+        $(this).selectText();
+      });
+     },
+
+    bind: function() {
+      var _this = this;
+
+      $('#get-data').on('click', function() {
+        _this.hideOutput();
+        _this.getData($('#latest-select').val());
+      });
+
+    },
+
+    hideOutput: function() {
+      $('.output').hide();
+    },
+
+    showOutput: function() {
+      $('.output').show();
+    },
+
+    getData: function(postId) {
+      var _this = this;
+      var data = {
+        'action': 'get_audio_post_data',
+        'postId': postId
+      };
+
+      $.ajax({
+        url: _this.vars.ajaxurl,
+        type: 'get',
+        data: data,
+        success: function(response, status) {
+          return _this.useData(response, status);
+        }
+      });
+
+    },
+
+    useData: function(response, status) {
```

Comment:
I feel like this function doesn't do anything specifically and unless it would be reused somewhere else I would put this code inside the `success` of the ajax call. Makes it more readable